src/getters/waitForGraphSync.ts

```diff
@@ -0,0 +1,65 @@
+import getGraphClient from '~/getters/getGraphClient'
+import {
+    GetMetadataDocument,
+    GetMetadataQuery,
+    GetMetadataQueryVariables,
+} from '~/generated/gql/network'
+
+const SESSION_STORAGE_KEY = 'StreamrHubLastBlockNumber'
+
+export const saveLastBlockNumber = (blockNumber: number): void => {
```

Comment:
I guess we can safely ignore situation where 2 different async workflows fight for the "last" spot and the smaller block number wins, eh?

If it's possible tho, let's ensure we don't take steps back.

Reply:
What can I do to prevent a situation where we would be awaiting an outdated block number is that on each loop iteration I'd grab the currently stored block number